### ISSUE TITLE
Error when trying to create executable

### ISSUE THREAD

**Issue by Rapid1898-code** [pyinstaller/pyinstaller#8267]
Hello - i try to create an exe-file on Windows with the following modules in the code:
Using the pyinstaller version: 6.3.0

This is the code:

```
import os
import sys
from dotenv import load_dotenv
from openai import OpenAI

if __name__ == '__main__':
  path = os.path.abspath(os.path.dirname(sys.argv[0]))  
  fn = os.path.join(path, ".env")
  load_dotenv(fn) 
  client = OpenAI(api_key = os.environ.get("CHATGPT_API_KEY"))
 
  resp = client.chat.completions.create(
    model = "gpt-3.5-turbo",
    # model = "gpt-3.5-turbo-16k",
    messages=[
      {"role": "system", "content": "You are a helpful assistant"},
      {"role": "user", "content": "What is the capital of Germany?"}
    ],
  )

print(resp.usage)
answer = resp.choices[0].message.content
print(answer)   
```

I try to compile the exe using this command:

`pyinstaller --onefile --collect-data frozendict translateText3.py`

But i get this error message when running the command:

```
(text) C:\DEV\Fiverr\ORDER\robalf\SOLtranslateTXT>pyinstaller --onefile --collect-data frozendict translateText3.py
308 INFO: PyInstaller: 6.3.0
308 INFO: Python: 3.9.7
320 INFO: Platform: Windows-10-10.0.19045-SP0
321 INFO: wrote C:\DEV\Fiverr\ORDER\robalf\SOLtranslateTXT\translateText3.spec
327 WARNING: collect_data_files - skipping data collection for module 'frozendict' as it is not a package.
328 INFO: Extending PYTHONPATH with paths
['C:\\DEV\\Fiverr\\ORDER\\robalf\\SOLtranslateTXT']
553 INFO: checking Analysis
553 INFO: Building Analysis because Analysis-00.toc is non existent
554 INFO: Initializing module dependency graph...
555 INFO: Caching module graph hooks...
572 INFO: Analyzing base_library.zip ...
1496 INFO: Loading module hook 'hook-heapq.py' from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\PyInstaller\\hooks'...
1697 INFO: Loading module hook 'hook-encodings.py' from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\PyInstaller\\hooks'...
2856 INFO: Loading module hook 'hook-pickle.py' from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\PyInstaller\\hooks'...
3808 INFO: Caching module dependency graph...
3960 INFO: Running Analysis Analysis-00.toc
3961 INFO: Looking for Python shared library...
3968 INFO: Using Python shared library: C:\Users\WRSPOL\AppData\Local\Programs\Python\Python39\python39.dll
3969 INFO: Analyzing C:\DEV\Fiverr\ORDER\robalf\SOLtranslateTXT\translateText3.py
4102 INFO: Loading module hook 'hook-platform.py' from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\PyInstaller\\hooks'...
4173 INFO: Processing pre-safe import module hook urllib3.packages.six.moves from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\PyInstaller\\hooks\\pre_safe_import_module\\hook-urllib3.packages.six.moves.py'.
5093 INFO: Loading module hook 'hook-certifi.py' from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\stdhooks'...
5546 INFO: Loading module hook 'hook-pywintypes.py' from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\stdhooks'...
5778 INFO: Loading module hook 'hook-sysconfig.py' from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\PyInstaller\\hooks'...
6000 INFO: Loading module hook 'hook-pythoncom.py' from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\stdhooks'...
6106 INFO: Processing pre-safe import module hook win32com from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\pre_safe_import_module\\hook-win32com.py'.
6209 INFO: Loading module hook 'hook-win32com.py' from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\stdhooks'...
6934 INFO: Loading module hook 'hook-multiprocessing.util.py' from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\PyInstaller\\hooks'...
6996 INFO: Loading module hook 'hook-xml.py' from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\PyInstaller\\hooks'...
C:\DEV\.venv\text\lib\site-packages\mysql\connector\abstracts.py:130: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if group is 'connector_python':
C:\DEV\.venv\text\lib\site-packages\mysql\connector\optionfiles.py:98: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if group is 'connector_python':
7843 INFO: Loading module hook 'hook-pydantic.py' from 'C:\\DEV\\.venv\\text\\lib\\site-packages\\_pyinstaller_hooks_contrib\\hooks\\stdhooks'...
Traceback (most recent call last):
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\utils\hooks\__init__.py", line 282, in get_module_attribute
    return _get_module_attribute(module_name, attr_name)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\isolated\_parent.py", line 420, in wrapped
    return call(function, *args, **kwargs)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\isolated\_parent.py", line 389, in call
    return isolated.call(function, *args, **kwargs)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\isolated\_parent.py", line 319, in call
    raise RuntimeError(f"Child process call to {function.__name__}() failed with:\n" + output)
RuntimeError: Child process call to _get_module_attribute() failed with:
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\isolated\_child.py", line 63, in run_next_command
    output = function(*args, **kwargs)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\utils\hooks\__init__.py", line 278, in _get_module_attribute
    return getattr(module, attr_name)
  File "C:\DEV\.venv\text\lib\site-packages\pydantic\__init__.py", line 363, in __getattr__
    return _getattr_migration(attr_name)
  File "C:\DEV\.venv\text\lib\site-packages\pydantic\_migration.py", line 302, in wrapper
    raise PydanticImportError(f'`{import_path}` has been removed in V2.')
pydantic.errors.PydanticImportError: `pydantic:compiled` has been removed in V2.

For further information visit https://errors.pydantic.dev/2.5/u/import-error

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\WRSPOL\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\WRSPOL\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\DEV\.venv\text\Scripts\pyinstaller.exe\__main__.py", line 7, in <module>
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\__main__.py", line 214, in _console_script_run
    run()
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\__main__.py", line 198, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\__main__.py", line 69, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\building\build_main.py", line 1071, in main
    build(specfile, distpath, workpath, clean_build)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\building\build_main.py", line 1011, in build
    exec(code, spec_namespace)
  File "C:\DEV\Fiverr\ORDER\robalf\SOLtranslateTXT\translateText3.spec", line 8, in <module>
    a = Analysis(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\building\build_main.py", line 470, in __init__
    self.__postinit__()
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\building\datastruct.py", line 184, in __postinit__
    self.assemble()
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\building\build_main.py", line 608, in assemble
    priority_scripts.append(self.graph.add_script(script))
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 268, in add_script
    self._top_script_node = super().add_script(pathname)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1160, in add_script
    self._process_imports(n)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 2591, in _process_imports
    target_modules = self._safe_import_hook(*import_info, **kwargs)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 437, in _safe_import_hook
    return super()._safe_import_hook(target_module_partname, source_module, target_attr_names, level, edge_attr)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 2048, in _safe_import_hook
    target_modules = self.import_hook(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1232, in import_hook
    target_package, target_module_partname = self._find_head_package(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1411, in _find_head_package
    target_package = self._safe_import_module(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 484, in _safe_import_module
    return super()._safe_import_module(module_basename, module_name, parent_package)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1792, in _safe_import_module
    self._process_imports(n)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 2591, in _process_imports
    target_modules = self._safe_import_hook(*import_info, **kwargs)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 437, in _safe_import_hook
    return super()._safe_import_hook(target_module_partname, source_module, target_attr_names, level, edge_attr)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 2257, in _safe_import_hook
    self.import_hook(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1271, in import_hook
    for target_submodule in self._import_importable_package_submodules(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1520, in _import_importable_package_submodules
    submodule = self._safe_import_module(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 484, in _safe_import_module
    return super()._safe_import_module(module_basename, module_name, parent_package)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1792, in _safe_import_module
    self._process_imports(n)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 2591, in _process_imports
    target_modules = self._safe_import_hook(*import_info, **kwargs)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 437, in _safe_import_hook
    return super()._safe_import_hook(target_module_partname, source_module, target_attr_names, level, edge_attr)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 2048, in _safe_import_hook
    target_modules = self.import_hook(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1232, in import_hook
    target_package, target_module_partname = self._find_head_package(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1411, in _find_head_package
    target_package = self._safe_import_module(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 484, in _safe_import_module
    return super()._safe_import_module(module_basename, module_name, parent_package)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1792, in _safe_import_module
    self._process_imports(n)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 2591, in _process_imports
    target_modules = self._safe_import_hook(*import_info, **kwargs)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 437, in _safe_import_hook
    return super()._safe_import_hook(target_module_partname, source_module, target_attr_names, level, edge_attr)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 2048, in _safe_import_hook
    target_modules = self.import_hook(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1232, in import_hook
    target_package, target_module_partname = self._find_head_package(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1411, in _find_head_package
    target_package = self._safe_import_module(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 484, in _safe_import_module
    return super()._safe_import_module(module_basename, module_name, parent_package)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1792, in _safe_import_module
    self._process_imports(n)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 2591, in _process_imports
    target_modules = self._safe_import_hook(*import_info, **kwargs)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 437, in _safe_import_hook
    return super()._safe_import_hook(target_module_partname, source_module, target_attr_names, level, edge_attr)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 2048, in _safe_import_hook
    target_modules = self.import_hook(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1232, in import_hook
    target_package, target_module_partname = self._find_head_package(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1411, in _find_head_package
    target_package = self._safe_import_module(
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 484, in _safe_import_module
    return super()._safe_import_module(module_basename, module_name, parent_package)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1792, in _safe_import_module
    self._process_imports(n)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 2591, in _process_imports
    target_modules = self._safe_import_hook(*import_info, **kwargs)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 368, in _safe_import_hook
    excluded_imports = self._find_all_excluded_imports(source_module.identifier)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\analysis.py", line 356, in _find_all_excluded_imports
    excluded_imports.update(module_hook.excludedimports)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\imphook.py", line 316, in __getattr__
    self._load_hook_module()
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\depend\imphook.py", line 383, in _load_hook_module
    self._hook_module = importlib_load_source(self.hook_module_name, self.hook_filename)
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\compat.py", line 581, in importlib_load_source
    mod_loader.exec_module(mod)
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "C:\DEV\.venv\text\lib\site-packages\_pyinstaller_hooks_contrib\hooks\stdhooks\hook-pydantic.py", line 22, in <module>
    is_compiled = get_module_attribute('pydantic', 'compiled') in {'True', True}
  File "C:\DEV\.venv\text\lib\site-packages\PyInstaller\utils\hooks\__init__.py", line 284, in get_module_attribute
    raise AttributeError(f"Failed to retrieve attribute {attr_name} from module {module_name}") from e
AttributeError: Failed to retrieve attribute compiled from module pydantic
```

How can i build the excecutable for this program?

**Comment by rokm** [comment]
What version of `pyinstaller-hooks-contrib` are you using? This should be fixed since `2023.5`.

**Comment by Rapid1898-code** [comment]
Yes i think this was the problem - after running

`pip install -U pyinstaller-hooks-contrib`

everything worked fine now
